src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -779,11 +779,9 @@
   [factset-id removed-pid-vid-pairs]
   (when-let [removed-pid-vid-pairs (seq removed-pid-vid-pairs)]
     (let [vids (map second removed-pid-vid-pairs)
-          fp-id-pairs (map vector
-                           (repeat factset-id )
-                           (map first removed-pid-vid-pairs))
+          rm-facts (map cons (repeat factset-id) removed-pid-vid-pairs)
```

Comment:
Actually I think either is fine, looks like you're just throwing the structure around it away on line 796